packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+        const absoluteUrl = await mainContainer.getAbsoluteUrl("");
+        return createSummarizerWithContainer(provider, absoluteUrl, runtimeFactory, { configProvider }, summaryVersion);
+    };
+    const summarize = async (summarizer: ISummarizer) => {
+        await provider.ensureSynchronized();
+        return summarizeNow(summarizer);
+    };
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    const createUnreferencedDataStore = async () => {
+        const handleKey = "handle";
+        const testDataObject = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
+        const testDataObjectId = testDataObject.id;
+
+        // Reference a datastore and summarize
+        mainDataStore._root.set(handleKey, testDataObject.handle);
+        const {
+            container: summarizingContainer1,
+            summarizer: summarizer1,
+        } = await loadSummarizerAndContainer();
+        await summarize(summarizer1);
+
+        // Unreference a datastore and summarize
+        mainDataStore._root.delete(handleKey);
+        const summaryVersion = (await summarize(summarizer1)).summaryVersion;
+        testDataObject._root.set("send while unreferenced", "op");
+        await provider.ensureSynchronized();
+        mainContainer.close();
+        summarizingContainer1.close();
+
+        return {
+            testDataObjectId,
+            summaryVersion,
+        };
+    };
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sessionExpiryTimeoutMs` above
+    it("Container loaded after sweep timeout expires can't send ops for tombstoned datastores", async () => {
+        const { testDataObjectId, summaryVersion } = await createUnreferencedDataStore();
+
+        // Wait a sweep worthy amount of time (all containers should have closed by now)
+        await delay(sweepTimeoutMs);
+
+        // Load a new container and summarizer based on the latest summary, summarize
+        const {
+            container: summarizingContainer2,
+            summarizer: summarizer2,
+        } = await loadSummarizerAndContainer(summaryVersion);
+
+        // Use the request pattern to get the testDataObject - this is unsafe and no one should do this in their
+        // production application
+        const testDataObject2 = await requestFluidObject<TestDataObject>(summarizingContainer2, testDataObjectId);
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer2);
+
+        // The request pattern should fail!
```

Comment:
This is testing changing the datastore not requesting it, right?

Reply:
Yes